src/components/text-input/TextInput.scss

```diff
@@ -1,55 +1,105 @@
 @import '../../styles/variables';
+@import '../../styles/inputs';
 
-/**************************************
- * Text Input
- **************************************/
+.bdl-TextInput-input {
+    @include common-typography;
+
+    height: 32px;
+    margin-top: 5px;
+
+    &[type="text"],
+    &[type="password"],
+    &[type="search"],
+    &[type="email"],
+    &[type="url"],
+    &[type="tel"],
+    &[type="number"] {
+        @include box-inputs;
+
+        width: 262px;
+
+        &:hover {
+            @include box-inputs-hover;
+        }
+
+        &:focus {
+            @include box-inputs-focus;
+        }
+
+        &:disabled {
+            background-color: $haze;
+            border-color: $efive;
+            box-shadow: none;
+            color: $bfive;
+
+            &:hover {
+                border-color: $efive;
+            }
+        }
+    }
+
+    // IE11 adds a new x to the inputs to clear it
+    &::-ms-clear {
+        display: none;
+    }
+
+    &::-webkit-input-placeholder,
+    &::-moz-placeholder,
+    &::-ms-input-placeholder {
+        color: $seesee;
+    }
+
+    /* Safari does't treat search input like a normal textfield */
+    &[type="search"] {
+        appearance: textfield;
+
+        &::-webkit-search-cancel-button,
+        &::-webkit-search-results-button {
+            display: none;
+        }
+    }
+
+    &[type="number"] {
+        appearance: textfield;
+
+        &::-webkit-outer-spin-button,
+        &::-webkit-inner-spin-button {
+            -webkit-appearance: none;
+            margin: 0;
+        }
+    }
+}
+
+.bdl-TextInput-container {
+    @include common-typography;
 
-.text-input-container {
     margin: 0 0 20px;
     position: relative;
 
-    .text-input-loading {
+    &.bdl-hasError {
+        .bdl-TextInput-input {
+            border-color: $amaranth;
+        }
+    }
+
+    .bdl-TextInput-loadingIndicator {
```

Comment:
All these `bdl-TextInput-FOO` styles here can/should be replaced with SCSS convention `&-FOO`

Reply:
How far would we want to go with this? For example, `${namespace}-${ComponentName}` is the top-level selector name? or just `${namespace}` with `-${ComponentName}` immediately after as a nested partial selector?

Reply:
Based on some feedback we might as well just keep full strings for searchability. Since concatenation can make that harder